cmake/HPX_SetupCUDA.cmake

```diff
@@ -6,57 +6,118 @@
 
 if(HPX_WITH_CUDA AND NOT TARGET Cuda::cuda)
 
-  find_package(CUDA REQUIRED)
+  if(HPX_WITH_CUDA_CLANG AND NOT (CMAKE_CXX_COMPILER_ID STREQUAL "Clang"))
+    hpx_error(
+      "To use Cuda Clang, please select Clang as your default C++ compiler"
+    )
+  endif()
+
+  # cuda_std_17 not recognized for previous versions
+  cmake_minimum_required(VERSION 3.18 FATAL_ERROR)
+
+  # Check CUDA standard
+  if(NOT DEFINED CMAKE_CUDA_STANDARD)
+    if(DEFINED CMAKE_CXX_STANDARD)
+      set(CMAKE_CUDA_STANDARD ${CMAKE_CXX_STANDARD})
+    else()
+      set(CMAKE_CUDA_STANDARD 17)
+    endif()
+  else()
+    if(CMAKE_CUDA_STANDARD LESS 17)
+      hpx_error(
+        "You've set CMAKE_CUDA_STANDARD to ${CMAKE_CUDA_STANDARD}, which is less than 17 (the minimum required by HPX)"
+      )
+    endif()
+  endif()
+
+  enable_language(CUDA)
+
   if(NOT HPX_FIND_PACKAGE)
     # The cmake variables are supposed to be cached no need to redefine them
     set(HPX_WITH_COMPUTE ON)
     hpx_add_config_define(HPX_HAVE_CUDA)
     hpx_add_config_define(HPX_HAVE_COMPUTE)
   endif()
-  # keywords for target_link_libraries (cuda)
-  set(CUDA_LINK_LIBRARIES_KEYWORD "PRIVATE")
 
-  if(HPX_WITH_CUDA_CLANG AND NOT (CMAKE_CXX_COMPILER_ID STREQUAL "Clang"))
-    hpx_error(
-      "To use Cuda Clang, please select Clang as your default C++ compiler"
-    )
+  # CUDA libraries used
+  add_library(Cuda::cuda INTERFACE IMPORTED)
+  # Toolkit targets like CUDA::cudart, CUDA::cublas, CUDA::cufft, etc. available
+  find_package(CUDAToolkit MODULE REQUIRED)
+  if(CUDAToolkit_FOUND)
+    target_link_libraries(Cuda::cuda INTERFACE CUDA::cudart)
+    target_link_libraries(Cuda::cuda INTERFACE CUDA::cublas)
   endif()
+  # Flag not working for CLANG CUDA
+  target_compile_features(Cuda::cuda INTERFACE cuda_std_${CMAKE_CUDA_STANDARD})
 
-  add_library(Cuda::cuda INTERFACE IMPORTED)
-  target_include_directories(Cuda::cuda INTERFACE ${CUDA_INCLUDE_DIRS})
+  # CUDA variables
+  set(CUDA_SEPARABLE_COMPILATION ON)
+  set(POSITION_INDEPENDENT_CODE ON)
+  # Add all the architectures to be targeted
+  if(NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
+    hpx_warn("CMAKE_CUDA_ARCHITECTURES not specified, using sm_60 by default.")
+    set(CMAKE_CUDA_ARCHITECTURES 60)
+  endif()
+  # keywords for target_link_libraries (cuda)
+  set(CUDA_LINK_LIBRARIES_KEYWORD "PRIVATE")
```

Comment:
I *think* this can go. As far as I can tell it shouldn't be required anymore since we don't use `FindCUDA`.